src/DynamoCoreWpf/UI/Prompts/PresetPrompt.xaml.cs

```diff
@@ -24,6 +23,36 @@ public PresetPrompt()
 
         void OK_Click(object sender, RoutedEventArgs e)
         {
+            if (this.Owner != null)
+            {
+                var dataContext = this.Owner.DataContext as DynamoViewModel;
+                 //get the preset names from workspace
+                if (dataContext != null && dataContext.Model.CurrentWorkspace.Presets.Any())
+                {
+                    if (dataContext.Model.CurrentWorkspace.Presets.Any(x => x.Name == Text))
+                    {
+                        var newDialog = new PresetOverwritePrompt()
+                        {
+                            Owner = this,
+                            WindowStartupLocation = WindowStartupLocation.CenterOwner,
+                            Text = Wpf.Properties.Resources.PresetOverwrite
+                        };
+
+                        if (newDialog.ShowDialog() != true)
+                        {                                                        
+                            e.Handled = true;
+                            return;
+                        }
+                        //If the dialog result is true, then remove the old preset
+                        else
+                        {
+                           var oldPreset = dataContext.Model.CurrentWorkspace.Presets.FirstOrDefault(x => x.Name == Text);
+                           dataContext.Model.CurrentWorkspace.Presets.Remove(oldPreset);
```

Comment:
Please rename `WorkspaceModel.RemoveState` to `WorkspaceModel.RemovePreset` and use that API instead.